bugsnag.go

```diff
@@ -88,7 +88,7 @@ func AutoNotify(rawData ...interface{}) {
 		severity := defaultNotifier.getDefaultSeverity(rawData, SeverityError)
 		state := HandledState{SeverityReasonHandledPanic, severity, true, ""}
 		rawData = append([]interface{}{state}, rawData...)
-		defaultNotifier.NotifySync(append(rawData, errors.New(err, 2), true)...)
+		defaultNotifier.NotifySync(errors.New(err, 2), true, rawData...)
```

Comment:
Is the stack trace definitely correct here? It looks like we skip the first two lines of the stacktrace with `errors.New(err, 2)` then later in [newEvent](https://github.com/bugsnag/bugsnag-go/blob/session-tracking-revert-notify-signature/event.go#L131) we seemingly skip a further 4 lines of the stacktrace.

I ask as the body of the `Notify` function above has been changed from:
 `return defaultNotifier.Notify(errors.New(err, 1), rawData...)`
to 
 `return defaultNotifier.Notify(err, rawData...)`
And I would have thought a similar change would be required here?

Reply:
That's right. This skip is to remove:

1. A line in `runtime/${arch}` (e.g. `runtime/asm_amd64`)
1. The line in `runtime.panic`

The four skips in `newEvent` are to remove:

1. The line in `newEvent`
1. The line in `notifier.NotifySync`
1. The line in `notifier.Notify`
1. The line in `bugsnag.Notify`

We do have a bit of testing to ensure we get the stackframes we expect. I've also verified this manually.

I'll add some comments around these numbers, because as you rightly point out it's not obvious.